### PR TITLE
explorer: Improve display of serum dex instructions and provide labels to market pairs

### PR DESCRIPTION
**Pull request by oJshua**
#### Problem
- Serum instructions are parsed as unknown. It would be nice to provide titles for the specific instructions.
- Serum accounts need labels to the specific market pairs.

#### Summary of Changes
- Added instruction table for Serum and displayed as title on InstructionCard
- Added labels for addresses of Serum markets

Fixes https://github.com/solana-labs/solana/issues/11979 https://github.com/solana-labs/solana/issues/11937

### CODE REVIEW

**Comment by jstarry** [comment]
Seeing some odd behavior here:
`/address/SRMuApVNdxXokk5GT7XD5cUUgXMBCoAz2LHeuAoKWRt/largest`

Screen Shot 2020-09-03 at 1 27 28 PM

- It's confusing that each of the accounts for a given market have the same label (eg "Serum: SRM / USDC" is the owner of "Serum: SRM / USDC")
- Looks like addresses should be left aligned in the owner column

Also, it looks like the token details page doesn't show the label yet, could you add that?

**Comment by oJshua** [comment]
@jstarry I'm not seeing the missing label on token details. Could you paste a link?

**Comment by jstarry** [comment]
Screen Shot 2020-09-04 at 12 41 11 AM

Some of the accounts you labeled seem to be token accounts. When loading the account details page for a token account, we don't show a row in the info for the label of that address despite it being labeled.

**Comment by oJshua** [comment]
@jstarry Ah, I see now. Thanks!

File: explorer/src/utils/tx.ts [hunk 1]
```diff
@@ -132,3 +138,41 @@ export function intoParsedTransaction(tx: Transaction): ParsedTransaction {
...
+  5: "Settle Funds",
+  6: "Cancel Order By Client Id",
+  7: "Disable Market",
+  8: "Sweep Fees",
+};
+
+export function parseSerumInstruction(
+  instruction: TransactionInstruction
+): ParsedSerumInstruction {
+  const code = instruction.data.slice(1, 5).readUInt32LE(0);
```
Comment (jstarry) [inline explorer/src/utils/tx.ts:166]: Can we wrap in a try catch in case this throws?

File: explorer/src/utils/tx.ts [hunk 2]
```diff
@@ -127,3 +133,33 @@ export function intoParsedTransaction(tx: Transaction): ParsedTransaction {
...
+export function parseSerumInstructionTitle(
+  instruction: TransactionInstruction
+): string {
+  try {
+    const code = instruction.data.slice(1, 5).readUInt32LE(0);
+
+    if (code in SERUM_CODE_LOOKUP) {
+      return SERUM_CODE_LOOKUP[code];
+    }
+  } catch (error) {}
```
Comment (jstarry) [inline explorer/src/utils/tx.ts:162]: It'd be nice to know when this fails in Sentry, could you report this error?

Comment (oJshua) [reply explorer/src/utils/tx.ts:162]: Yeah, was going to ask if we cared about that. Will add!

**Review by jstarry — APPROVED** [review]
Looks great!

**Pull request merged**